pandas/core/arrays/base.py

```diff
@@ -1044,7 +1044,7 @@ def factorize(self, na_sentinel: int = -1) -> tuple[np.ndarray, ExtensionArray]:
         arr, na_value = self._values_for_factorize()
 
         codes, uniques = factorize_array(
-            arr, na_sentinel=na_sentinel, na_value=na_value
+            arr, na_sentinel=na_sentinel, na_value=na_value, ignore_na=True
```

Comment:
is ignore_na just always going to be True?

Reply:
I think no - this part (EAs) is still to come.